File: shared-atlas-dart/lib/src/interfaces/cursor_update_payload.dart
/// Payload for cursor position update.
/// 光标位置更新的 Payload。
/// 
/// Event: `client.emit('cursor:update', payload)`
/// Event: `server.emit('cursor:update', payload)`
class CursorUpdatePayload {
  /// Note ID.
  /// 笔记ID。
  final String noteId;
  /// Current cursor position.
  /// 当前光标位置。
  final dynamic position;
  /// Current selection range.
  /// 当前选区范围。
  final dynamic selection;

  CursorUpdatePayload({
    required this.noteId,
    this.position,
    this.selection,
  });

  factory CursorUpdatePayload.fromJson(Map<String, dynamic> json) {
    return CursorUpdatePayload(
      noteId: json['noteId'] as String,
      position: json['position'] as dynamic,
      selection: json['selection'] as dynamic,
    );
  }
  Map<String, dynamic> toJson() {
    return {
      'noteId': this.noteId,
      'position': this.position,
      'selection': this.selection,
    };
  }
}

File: packages/shared-atlas-rust/src/interfaces/cursor_update_payload.rs
use serde::{Serialize, Deserialize};
use serde_json;

/// Payload for cursor position update.
/// 光标位置更新的 Payload。
/// 
/// Event: `client.emit('cursor:update', payload)`
/// Event: `server.emit('cursor:update', payload)`
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CursorUpdatePayload {
/// Note ID.
/// 笔记ID。
    pub note_id: String,
/// Current cursor position.
/// 当前光标位置。
    pub position: Option<serde_json::Value>,
/// Current selection range.
/// 当前选区范围。
    pub selection: Option<serde_json::Value>,
}